### PR TITLE
Remove remaining Cylc Review files

### PR DESCRIPTION
**Pull request by kinow**
Complement to 3f7a601ebc3ff8c37df65b5d6013edfdb476a8aa, removes the remaining files related to Cylc Review, as well as documentation mentioning it.

Checked if `TASK_STATUS_GROUPS` was used with the IDE, then with `grep`, and finally confirmed where it was used before with ol': `git log -S "TASK_STATUS_GROUPS" --patch`.

I think it should be safe to remove it too, as not used in the current code base. Hopefully not breaking anything in Travis-CI. Suite `five` and all unit tests running OK locally.

### CODE REVIEW

**Comment by kinow** [comment]
I believe this pull request is Cylc 8 only, without the need to backport to 7.8 branch.

**Comment by kinow** [comment]
Found while reviewing External Triggers following up discussion in Riot.

File: tests/lib/bash/test_header [hunk 1]
```diff
@@ -105,25 +105,6 @@
 #         set CYLC_TEST_HOST and CYLC_TEST_OWNER for remote job tests.
 #     port_is_busy $PORT
 #         Return 0 if $PORT is busy or 1 if $PORT is not busy.
-#     cylc_ws_init $NS $UTIL
```
Comment (sadielbartholomew) [inline tests/lib/bash/test_header]: We can also now get rid of the ``port_is_busy`` method, which was only used across the test suite inside this ``cylc_ws_init`` method. (And now I have noticed it, we should drop the ``# tags`` on new line 580, just above where that method is defined, which was clearly for the aid of some developer who neglected to remove it. From ``git blame`` I can see that the developer in question was in fact myself :grimacing:.)

Comment (kinow) [reply tests/lib/bash/test_header]: Good spot, thanks for reviewing it so quickly. Updating PR in a few moments. Thanks @sadielbartholomew !

Comment (kinow) [reply tests/lib/bash/test_header]: Hopefully removed the right lines [here](https://github.com/cylc/cylc/pull/3024/commits/860b1b131ab1456b4df3bd7409a16037f6050d45)

Comment (sadielbartholomew) [reply tests/lib/bash/test_header]: Excellent, thanks. Given the time in NZ I imagine you are no longer online so I have given two Travis CI jobs that were failing on flaky tests a kick. Once they pass I'll get this merged.

**Pull request merged**